### ISSUE TITLE
Poorly documented, library code broken, please step it up. 

### ISSUE THREAD

**Issue by fxfactorial** [PagerDuty/go-pagerduty#170]
This library is surprisingly poorly documented, no examples around at all. I have to dig in the source code, this defeats the whole purpose of having a library and I have no idea the structs represent. 

```
type CreateIncidentOptions struct {
	Type             string        `json:"type"`
	Title            string        `json:"title"`
	Service          *APIReference `json:"service"`
	Priority         *APIReference `json:"priority,omitempty"`
	IncidentKey      string        `json:"incident_key,omitempty"`
	Body             *APIDetails   `json:"body,omitempty"`
	EscalationPolicy *APIReference `json:"escalation_policy,omitempty"`
}
```

What is an APIReference? or an APIDetail? No examples. 

EDIT: Searched in other people's code repos and now get the same error as mentioned in this ticket https://github.com/PagerDuty/go-pagerduty/issues/165, the fix of which was done eight days ago.

**Comment by stmcallister** [comment]
Sorry about the delay on this! Fix #165 has been merged.

Also, thank you for the feedback on samples. I agree. :) I've been fumbling around the source code myself to figure out how things work, which isn't how it should be. I'll make sure we get some samples posted in a repo soon.

**Comment by fxfactorial** [comment]
Thank you, appreciate the response. I will try this later today and ping back if more issues.
